trino-s3-proxy/src/test/java/io/trino/s3/proxy/server/testing/ManagedS3MockContainer.java

```diff
@@ -28,24 +33,58 @@
 
 public class ManagedS3MockContainer
 {
-    private final S3MockContainer container;
+    private final MinIOContainer container;
+
+    private static final String CONFIG = """
+            {
+                "version": "10",
+                "aliases": {
+                    "local": {
+                        "url": "http://localhost:9000",
+                        "accessKey": "%s",
+                        "secretKey": "%s",
+                        "api": "S3v4",
+                        "path": "auto"
+                    }
+                }
+            }
+            """;
 
     @Retention(RUNTIME)
     @Target({FIELD, PARAMETER, METHOD})
     @BindingAnnotation
     public @interface ForS3MockContainer {}
 
     @Inject
-    public ManagedS3MockContainer(@ForS3MockContainer String initialBuckets)
+    public ManagedS3MockContainer(@ForS3MockContainer String initialBuckets, @ForS3MockContainer Credential credential)
     {
-        container = new S3MockContainer("3.8.0");
+        Transferable transferable = Transferable.of(CONFIG.formatted(credential.accessKey(), credential.secretKey()));
+        container = new MinIOContainer("minio/minio:RELEASE.2023-09-04T19-57-37Z")
```

Comment:
I see on the `trino-testing` side that these get pinned to specific versions too, but get bumped every now and then:
- Any reason to stick to this specific version?
- Should we move these to two constants? `MINIO_IMAGE_NAME` and `MINIO_IMAGE_TAG`?